### PR TITLE
planner: rename some variables for easy understanding

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve?
Some variables related to plan cache is hard to understand.

### What is changed and how it works?
Rename some variables related to plan cache for easy understanding.

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
'None'
```

### CODE REVIEW

File: expression/util.go [hunk 1]
```diff
@@ -905,20 +905,33 @@ func ContainCorrelatedColumn(exprs []Expression) bool {
...
+// MaybeOverOptimized4PlanCache used to check whether an optimization can work
+// for the statement when we enable the plan cache.
+// In some situations, some optimizations maybe over-optimize and cache an
+// overOptimized plan. The cached plan may not get the correct result when we
+// reuse the plan for other statements. So we need to do the check here.
+// The check includes the following aspects:
+// 1. Whether the plan cache switch is enable.
+// 2. Whether the expressions contain a lazy constant.
+// TODO: Do more careful check here.
+func MaybeOverOptimized4PlanCache(ctx sessionctx.Context, exprs []Expression) bool {
```
Comment (qw4990) [inline expression/util.go:917]: How about adding a simple case to explain what is `over-optimized`?
For example, `c>=$a and c<=$b` --> `c==$a` ($b is abandoned after optimization)

**Comment by rebelice** [comment]
almost LGTM, the same comment as qw4990

**Comment by Reminiscent** [comment]
@qw4990 @rebelice PTAL

**Review by qw4990 — APPROVED** [review]

**Review by rebelice — APPROVED** [review]

**Pull request merged**